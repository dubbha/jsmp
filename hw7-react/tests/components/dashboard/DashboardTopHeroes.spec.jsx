import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DashboardTopHeroes, mapStateToProps, mapDispatchToProps }
  from 'components/dashboard/DashboardTopHeroes';

jest.mock('components/heroes/heroes.actions', () => ({
  updateHero: jest.fn(),
}));

jest.mock('components/heroes/heroes.selectors', () => ({
  heroesSelector: state => state.heroes,
}));

describe('DashboardTopHeroes', () => {
  let props;

  beforeEach(() => {
    props = {
      heroes: [
        { id: 1, name: 'Optimus Prime' },
        { id: 2, name: 'Optimus Duos' },
        { id: 3, name: 'Optimus Tres' },
      ],
      updateHero: jest.fn(),
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render dashboard top heroes list', () => {
    const wrapper = shallow(<DashboardTopHeroes {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const state = {
      heroes: [
        { id: 4, name: 'Optimus Four' },
        { id: 5, name: 'Optimus Five' },
      ]
    };
    expect(mapStateToProps(state)).toEqual({
      heroes: [
        { id: 4, name: 'Optimus Four' },
        { id: 5, name: 'Optimus Five' },
      ]
    });
  });

  it('should map dispatch to props', () => {
    const dispatch = jest.fn();
    const hero = { id: 1, name: 'Primus' };
    expect(mapDispatchToProps(dispatch).updateHero(hero)).toEqual(dispatch(hero));
  });
});
