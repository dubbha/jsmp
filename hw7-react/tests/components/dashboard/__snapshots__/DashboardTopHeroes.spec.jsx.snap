// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTopHeroes should render dashboard top heroes list 1`] = `
<div>
  <h3>
    Top Heroes
  </h3>
  <div
    className="grid grid-pad"
  >
    <DashboardTopHero
      hero={
        Object {
          "id": 1,
          "name": "Optimus Prime",
        }
      }
    />
    <DashboardTopHero
      hero={
        Object {
          "id": 2,
          "name": "Optimus Duos",
        }
      }
    />
    <DashboardTopHero
      hero={
        Object {
          "id": 3,
          "name": "Optimus Tres",
        }
      }
    />
  </div>
</div>
`;
